pandas/core/reshape/pivot.py

```diff
@@ -534,28 +534,38 @@ def crosstab(
     b      0  1  0
     c      0  0  0
     """
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
+    if values is not None and aggfunc is None:
+        raise ValueError("values cannot be used without an aggfunc.")
 
     index = com.maybe_make_list(index)
     columns = com.maybe_make_list(columns)
 
-    rownames = _get_names(index, rownames, prefix="row")
-    colnames = _get_names(columns, colnames, prefix="col")
-
     common_idx = _get_objs_combined_axis(index + columns, intersect=True, sort=False)
 
-    data = {}
-    data.update(zip(rownames, index))
-    data.update(zip(colnames, columns))
-
-    if values is None and aggfunc is not None:
-        raise ValueError("aggfunc cannot be used without values.")
+    rownames = _get_names(index, rownames, prefix="row")
+    colnames = _get_names(columns, colnames, prefix="col")
 
-    if values is not None and aggfunc is None:
-        raise ValueError("values cannot be used without an aggfunc.")
+    # We create our own mapping of row and columns names
```

Comment:
this is too complicated; rather I would just factorize the names this creates the names and provides a reverse mapping (see algorithms.py) for usage.

Reply:
Thanks for the pointer! I will replace `_get_duplicate_count()` with _value_counts_arraylike() from algorithms.py. 

Looked at factorize, but unless I am missing something we would still need to build some type of mapping with unique names (the labels from factorize are not unique). 

If _build_names_mapper() is too complicated and alternative is to give or append a number to each name, but in case of an exception in pivot_table() mentioning the columns that would be confusing for users. 

Any other ideas that I am overlooking?

Reply:
Updated the code. Is this still too complicated? If yes, we could blindly add a number to each row and column name. The disadvantage is that it will confuse users if any errors are raised while creating the DataFrame or calling pivot_table().